game_engine/core/HaliteImpl.cpp

```diff
@@ -524,6 +532,8 @@ void HaliteImpl::update_player_stats() {
             player_stats.turn_deposited.push_back(0);
         }
     }
+	// Used to track the current turn number inside Event::update_stats
+	game.game_statistics.turn_number = game.turn_number;
```

Comment:
Turn numbering is confusing and maybe I'm getting it wrong; but it seems like this is introducing an off by one for the recorded turn numbers. Since the turn number is being set at the end of the turn here and then used throughout the next turn.

I think the correct place to do this is around line 143 when the logging turn number is also set. That way the turn number is set before the turn is processed.